Feature/select tag improvements

siemensikkema opened the pull request:
- [x] Add public `OptionRepresentable` (modified from internal `Selectable`)
- [x] Document

steffendsommer commented:
Looks good to me - @siemensikkema what could be a good example of having to pass in the label?

siemensikkema commented:
I'm using it for a type that is not `Submittable` and I'm not creating any `Field`s. One could argue that we don't need to support that though.

siemensikkema commented:
@steffendsommer I removed the label parameter for the select tag (it's currently in https://github.com/nodes-vapor/submissions/tree/feature/enable-specifying-label-in-leaf in case we need it) and added documentation. One extra small thing: I'm treating the "multiple" keyword in the file tag as boolean as per the spec.
